packages/contracts-bedrock/test/vendor/Initializable.t.sol

```diff
@@ -225,242 +235,273 @@ contract Initializer_Test is Bridge_Initializer {
-                initializedSlotVal: deploy.loadInitializedSlot("DataAvailabilityChallengeProxy")
+                initCalldata: abi.encodeCall(dataAvailabilityChallenge.initialize, (address(0), 0, 0, 0, 0))
+            })
+        );
+        // AnchorStateRegistry
+        contracts.push(
+            InitializeableContract({
+                name: "AnchorStateRegistry",
+                target: address(anchorStateRegistry),
+                initCalldata: abi.encodeCall(
+                    anchorStateRegistry.initialize,
+                    (new AnchorStateRegistry.StartingAnchorRoot[](1), SuperchainConfig(address(0)))
+                )
             })
         );
+        // AnchorStateRegistryProxy
+        contracts.push(
+            InitializeableContract({
+                name: "AnchorStateRegistryProxy",
+                target: address(anchorStateRegistry),
+                initCalldata: abi.encodeCall(
+                    anchorStateRegistry.initialize,
+                    (new AnchorStateRegistry.StartingAnchorRoot[](1), SuperchainConfig(address(0)))
+                )
+            })
+        );
+
+        // Nicknamed contracts.
+        nicknames["OptimismPortal2Proxy"] = "OptimismPortalProxy";
     }
 
     /// @notice Tests that:
-    ///         1. All `Initializable` contracts in `src/L1` and `src/L2` are accounted for in the `contracts` array.
-    ///         2. The `_initialized` flag of each contract is properly set to `1`, signifying that the
-    ///            contracts are initialized.
-    ///         3. The `initialize()` function of each contract cannot be called more than once.
+    ///         1. All `Initializable` contracts in `src/` (except periphery) are accounted for in `contracts`.
+    ///         2. The `_initialized` flag of each contract is properly set.
+    ///         3. The `initialize()` function of each contract cannot be called again.
     function test_cannotReinitialize_succeeds() public {
-        // Ensure that all L1, L2 `Initializable` contracts are accounted for, in addition to
-        // OptimismMintableERC20FactoryImpl, OptimismMintableERC20FactoryProxy, OptimismPortal2,
-        // DisputeGameFactoryImpl, DisputeGameFactoryProxy, DelayedWETHImpl, DelayedWETHProxy.
-        // Omitting OptimismSuperchainERC20 due to using OZ v5 Initializable.
-        assertEq(_getNumInitializable(), contracts.length);
+        // Get names of all contracts in the src directory.
+        string[] memory excludes = new string[](6);
+        excludes[0] = "src/L1/SystemConfigInterop.sol";
+        excludes[1] = "src/L1/OptimismPortalInterop.sol";
+        excludes[2] = "src/L2/OptimismSuperchainERC20.sol";
+        excludes[3] = "src/periphery/*";
+        // TODO: Actually include these contracts once we fix the deployment script.
+        excludes[4] = "src/dispute/FaultDisputeGame.sol";
+        excludes[5] = "src/dispute/PermissionedDisputeGame.sol";
```

Comment:
What is the rationale for these exclusions, partocularly the first 4?

Reply:
I will add comments for each